lua/entities/gmod_wire_expression2/core/custom/cfc_vector2_NumberOperands.lua

```diff
@@ -0,0 +1,537 @@
+        return 0
+    end
+    return {
+        numIsInf(vectah[1]),
+        numIsInf(vectah[2])
+    }
+end
+
+e2function vector2 vector2:isinf()
+    local function numIsInf(number)
+        if number == inf then return 1 end
+        if number == -inf then return -1 end
+        return 0
+    end
+    return {
+        numIsInf(this[1]),
+        numIsInf(this[2])
+    }
+end
+
+--- Returns true (1) if given value is not a number (NaN); otherwise false (0).
+e2function vector2 isnan(vector2 vectah)
+    local function isNaNNum(number)
+        return (value ~= value) and 1 or 0
+    end
+    return {
+        isNaNNum(vectah[1]),
+        isNaNNum(vectah[2])
+    }
+end
+e2function vector2 vector2:isnan()
+    local function isNaNNum(number)
+        return (value ~= value) and 1 or 0
+    end
+    return {
+        isNaNNum(this[1]),
+        isNaNNum(this[2])
+    }
+end
+
+--[[************************************************************************]]--
+
+__e2setcost(2) -- approximation
+
+e2function vector2 abs(vector2 vectah)
+    return {
+        abs(vectah[1]),
+        abs(vectah[2])
+    }
+end
+e2function vector2 vector2:abs()
+    return {
+        abs(this[1]),
+        abs(this[2])
+    }
+end
+
+--- rounds towards zero
+e2function vector2 int(vector2 vectah)
+    local function intNum(number)
+        if rv1 >= 0 then return floor(rv1) else return ceil(rv1) end
+    end
+    return {
+        intNum(vectah[1]),
+        intNum(vectah[2])
+    }
+end
+e2function vector2 vector2:int()
+    local function intNum(number)
+        if rv1 >= 0 then return floor(rv1) else return ceil(rv1) end
+    end
+    return {
+        intNum(this[1]),
+        intNum(this[2])
+    }
+end
+
+--- returns the fractional part. (frac(-1.5) == 0.5 & frac(3.2) == 0.2)
+e2function vector2 frac(vector2 vectah)
+    local function fracnum(number)
+        if rv1 >= 0 then return rv1 % 1 else return rv1 % -1 end
+    end
+    return {
+        fracnum(vectah[1]),
+        fracnum(vectah[2])
+    }
+end
+
+e2function vector2 vector2:frac()
+    local function fracnum(number)
+        if rv1 >= 0 then return rv1 % 1 else return rv1 % -1 end
+    end
+    return {
+        fracnum(this[1]),
+        fracnum(this[2])
+    }
+end
+
+e2function vector2 vector2:wrap(vector2 wrapby)
+    return {
+        (this[1] + wrapby[1]) % (wrapby[1] * 2) - wrapby[1],
+        (this[2] + wrapby[2]) % (wrapby[2] * 2) - wrapby[2]
+    }
+end
+
+e2function vector2 wrap(vector2 wrapped, vector2 wrapby)
+    return {
+        (wrapped[1] + wrapby[1]) % (wrapby[1] * 2) - wrapby[1],
+        (wrapped[2] + wrapby[2]) % (wrapby[2] * 2) - wrapby[2]
+    }
+end
+
+--[[************************************************************************]]--
+
+__e2setcost(2) -- approximation
+
+e2function vector2 sqrt(vector2 vectah)
+    return {
+        vectah[1] ^ (1 / 2),
+        vectah[2] ^ (1 / 2)
+    }
+end
+e2function vector2 vector2:sqrt()
+    return {
+        this[1] ^ (1 / 2),
```

Comment:
```suggestion
        sqrt(this[1]),
```

You defined it above but never used it.